trino/client.py

```diff
@@ -684,6 +686,27 @@ def _verify_extra_credential(self, header):
             raise ValueError(f"only ASCII characters are allowed in extra credential '{key}'")
 
 
+class ResultDownloader():
+    def __init__(self):
+        self.queue: queue.Queue = queue.Queue()
```

Comment:
According to [the docs](https://docs.python.org/3/library/queue.html#queue.Queue), the default `maxsize` is zero, which means unbounded. When I was testing [similar changes](https://github.com/trinodb/trino-go-client/pull/57#discussion_r977541484) in the Go driver, it turned out that the queue size didn't matter, so even a queue of size 1 is fine. Did you test different queue sizes, and if not, can you check if `maxsize` greater than 1 makes a notable difference?